widgets/src/main/java/org/pentaho/gwt/widgets/client/utils/ElementUtils.java

```diff
@@ -285,4 +288,29 @@ public static native String getComputedStyle( Element ele, String styleName )/*-
     return "";
   }-*/;
 
+  /**
+   * Set's a widget's ARIA label.
+   * <p>If the provided label widget does not have an identifier, a unique one is assigned to it.</p>
+   * @param widget The widget.
+   * @param label The label.
+   */
+  public static void setAriaLabelledBy( Widget widget, Label label ) {
+    Property.LABELLEDBY.set( widget.getElement(), Id.of( ensureId( label ) ) );
+  }
+
+  /**
+   * Ensures that a given widget has an identifier, assigning it a unique one, if necessary.
+   * @param widget The widget.
+   * @return The widget's identifier.
+   */
+  public static String ensureId( Widget widget ) {
+    com.google.gwt.user.client.Element element = widget.getElement();
+    String id = element.getId();
+    if ( id == null || id.equals( "" ) ) {
```

Comment:
StringUtils.isBlank( id ) can be used here

Reply:
JFYI - https://www.educative.io/answers/what-is-stringutilsisblank-in-java

Reply:
null == id gives better readability which we are comparing with first than what we are comparing with. OK with both.
assume a case
if (str == "a"){
} else if (str == "aa") {
} else if ((str == "aaaa") {
can be also
if ("a" == str){
} else if ("aa" == str) {
} else if (("aaaa" == str) {

Reply:
Regarding `isBlank`. When I wrote this, I was a bit uneasy about what libraries I could use for GWT code (there are some Java classes which cannot be used). We do have a `StringUtils.isEmpty()` clone (on `org.pentaho.gwt.widgets.client.utils.string` of the Apache library) which can be used. Thanks!

Reply:
Regarding the operator order, I do not like that syntax. Was it with you that I talked about this, the other day? If so, did you find the Sonar rule you said existed?

From my perspective, the way you propose hurts readability. The only cases in which I invert the order is when using the `.equals( . )` method, to ensure that the left-hand side cannot be `null`, thus favoring the use of constants on the left.

Reply:
Addressed with `StringUtils.isEmpty(String)`.